src/index.js

```diff
@@ -39,32 +46,84 @@ function onLoadData() {
       destinations = values[1].destinations;
       traveler = values[2]
       let newTraveler = generateTraveler()
-      // console.log(newTraveler.pastTrips());
 
       onLoadDisplay(newTraveler, destinations)
+      // console.log(newTraveler)
     })
 }
 
+function fetchUserData(travelerID) {
+  let user = api.fetchOneTraveler(travelerID)
+  console.log(user)
+}
+
 function onLoadDisplay(traveler, destinations) {
   updateDom.welcomeMessage(traveler);
   updateDom.updatePastTrips(traveler);
   updateDom.updatePresentTrips(traveler);
   updateDom.updateFutureTrips(traveler);
   updateDom.updatePendingTrips(traveler);
   updateDom.generateDestinationList(destinations);
+  updateDom.displayMoneySpent(traveler);
 
 }
 
 function submitTrip() {
   if(validateDateEntry() && validateDuration() && validateTravelers() && validateDestination() === true) {
-    let newTrip = generateNewTrip()
-    //post the new trip here!!!
+    let trip = generateNewTrip()
+    postTrip()
     errorMessage.classList.add('hidden')
   } else {
     errorMessage.classList.remove('hidden')
   }
 }
 
+function generateNewTrip() {
+  let newTrip = {
+    id: Math.floor((Math.random() * 100) + 50),
+    userID: userID,
+    destinationID: parseInt(dropdown.value),
+    travelers: parseInt(travelerInput.value),
+    date: dateInput.value,
+    duration: parseInt(durationInput.value),
+    status: 'pending',
+    suggestedActivities: []
+  }
+  convertTripForPost(newTrip)
+  return newTrip
+}
+
+function convertTripForPost(trip) {
+  requestedTrip = {
+    id: Math.floor((Math.random() * 100) + 50),
```

Comment:
My guess is that because this number is within a range, the new id you are giving it might already exist on the api. So even though you are randomizing it, it likely is a number that has already been posted (hence why it worked sometimes for you).  To fix this, inside of convertTripForPost, change id to be `Date.now()`:

  ```
requestedTrip = {
    id: Date.now(),
    userID: +trip.userID,
    destinationID: +trip.destinationID,
    travelers: +trip.travelers,
    date: dateInput.value,
    duration: +trip.duration,
    status: 'pending',
    suggestedActivities: []
  }
```

Side note, once you have this working, you could probably refactor this and remove this `convertTripForPost` function since your `generateNewTrip` is already doing almost exactly the same thing.